packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -104,24 +171,317 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+    ]
+  );
+
+  const onTextFieldKeyDown = useCallback(
     (ev: React.KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>) => {
       // Uses KeyCode 229 and which code 229 to determine if the press of the enter key is from a composition session or not (Safari only)
       if (ev.nativeEvent.isComposing || ev.nativeEvent.keyCode === 229 || ev.nativeEvent.which === 229) {
         return;
       }
+      if (ev.key === 'ArrowUp') {
+        ev.preventDefault();
+        /* @conditional-compile-remove(mention) */
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? mentionSuggestions.length - 1
+              : Math.max(activeSuggestionIndex - 1, 0);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      } else if (ev.key === 'ArrowDown') {
+        ev.preventDefault();
+        /* @conditional-compile-remove(mention) */
+        if (mentionSuggestions.length > 0) {
+          const newActiveIndex =
+            activeSuggestionIndex === undefined
+              ? 0
+              : Math.min(activeSuggestionIndex + 1, mentionSuggestions.length - 1);
+          setActiveSuggestionIndex(newActiveIndex);
+        }
+      }
       if (ev.key === 'Enter' && (ev.shiftKey === false || !supportNewline)) {
         ev.preventDefault();
+
+        // If we are looking up a mention, select the focused suggestion
+        /* @conditional-compile-remove(mention) */
+        if (mentionSuggestions.length > 0 && activeSuggestionIndex !== undefined) {
+          const selectedMention = mentionSuggestions[activeSuggestionIndex];
+          if (selectedMention) {
+            onSuggestionSelected(selectedMention);
+            return;
+          }
+        }
+
         onEnterKeyDown && onEnterKeyDown();
       }
       onKeyDown && onKeyDown(ev);
     },
-    [onEnterKeyDown, onKeyDown, supportNewline]
+    [
+      onEnterKeyDown,
+      onKeyDown,
+      supportNewline,
+      /* @conditional-compile-remove(mention) */
+      mentionSuggestions,
+      /* @conditional-compile-remove(mention) */
+      activeSuggestionIndex,
+      /* @conditional-compile-remove(mention) */
+      onSuggestionSelected
+    ]
   );
 
+  /* @conditional-compile-remove(mention) */
+  const debouncedQueryUpdate = useRef(
+    useDebouncedCallback(async (query: string) => {
+      const suggestions = (await mentionLookupOptions?.onQueryUpdated(query)) ?? [];
+      if (suggestions.length === 0) {
+        setActiveSuggestionIndex(undefined);
+      } else if (activeSuggestionIndex === undefined) {
+        setActiveSuggestionIndex(0);
+      }
+      updateMentionSuggestions(suggestions);
+    }, 500)
+  ).current;
+
+  /* @conditional-compile-remove(mention) */
+  useEffect(() => {
+    return () => {
+      debouncedQueryUpdate.cancel();
+    };
+  }, [debouncedQueryUpdate]);
+
+  /* @conditional-compile-remove(mention) */
+  const handleOnChange = async (
```

Comment:
Any special reason we use async for this function? I don't see any await in the function body

Reply:
Wrap this with useCallback btw, avoid creating new instance everytime in rendering

Reply:
This change will be done in `vhuseinova/mention-update`